File: examples/devicesTutorial/robotcomponents/wam/WAMtest.cpp

#include <cisstCommon/cmnConstants.h>
#include <cisstDevices/can/devRTSocketCAN.h>
#include <cisstDevices/robotcomponents/manipulators/Barrett/devWAM.h>

#include <cisstMultiTask/mtsTaskManager.h>

#include <cisstOSAbstraction/osaGetTime.h>

int main(){

  mtsTaskManager* taskManager = mtsTaskManager::GetInstance();

  devRTSocketCAN can( "rtcan0", devCAN::RATE_1000 );
  vctDynamicVector<double> qinit(7, 0.0);
  qinit[1] = -cmnPI_2;
  qinit[3] =  cmnPI;

  devWAM wam( "WAM", 0.002, OSA_CPU1, &can, qinit );
  wam.Configure();
    
  taskManager->AddComponent( &wam );
  taskManager->CreateAll();
  taskManager->StartAll();

  getchar();

  return 0;
}

File: cisst/cisstDevices/examples/robotcomponents/scripts/WAMtest.py
from RobotComponents import *

cmnLogger.SetMask( CMN_LOG_ALLOW_ALL )
cmnLogger.SetMaskFunction( CMN_LOG_ALLOW_ALL )
cmnLogger.SetMaskDefaultLog( CMN_LOG_ALLOW_ALL )

taskManager = mtsManagerLocal.GetInstance()

rtcan = devRTSocketCAN( 'rtcan0', devCAN.RATE_1000 )

WAMqinit = array( [ 0.0, -pi/2, 0.0, pi, 0.0, 0.0, 0.0 ] )
WAM = CreateWAM( 'WAM', rtcan, WAMqinit )
WAM.Configure()
taskManager.AddComponent( WAM )

StartTasks()

stdin.readline()

StopTasks()
